packages/azure-http-specs/specs/azure/resource-manager/operation-templates/lro.tsp

```diff
@@ -0,0 +1,262 @@
+    final-state-via: location
+    
+    Expected verb: POST
+    Expected path: /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.ResourceManager.OperationTemplates/lroResources/lro/upload
+    Expected query parameter: api-version=2023-12-01-preview
+    Expected request body:
+    ```json
+    {
+      "content": "My File Content.",
+      "filename": "myfile"
+    }
+    ```
+    Expected response status code: 202
+    Expected response headers: 
+      - Azure-AsyncOperation={endpoint}/subscriptions/00000000-0000-0000-0000-000000000000/locations/eastus/lro_post/aao
+      - Location={endpoint}/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/lro_post/location
+    Expected no response body
+    
+    Whether you do polling through AAO, Location or combined, first one will respond with provisioning state "InProgress", second one with "Succeeded".
+    
+    AAO first poll.
+    Expected verb: GET
+    Expected URL: {endpoint}/subscriptions/00000000-0000-0000-0000-000000000000/locations/eastus/lro_post/aao
+    Expected status code: 200
+    Expected response body:
+    ```json
+    {
+      "id": "/subscriptions/00000000-0000-0000-0000-000000000000/locations/eastus/lro_post/aao",
+      "name": "aao",
+      "status" : "InProgress",
+      "startTime": "2024-11-08T01:41:53.5508583+00:00"
+    }
+    ```
+    
+    AAO second poll.
+    Expected verb: GET
+    Expected URL: {endpoint}/subscriptions/00000000-0000-0000-0000-000000000000/locations/eastus/lro_post/aao
+    Expected status code: 200
+    Expected response body:
+    ```json
+    {
+      "id": "/subscriptions/00000000-0000-0000-0000-000000000000/locations/eastus/lro_post/aao",
+      "name": "aao",
+      "status" : "Succeeded",
+      "startTime": "2024-11-08T01:41:53.5508583+00:00",
+      "endTime": "2024-11-08T01:42:41.5354192+00:00"
+    }
+    ```
+    
+    Location first poll.
+    Expected verb: GET
+    Expected URL: {endpoint}/subscriptions/00000000-0000-0000-0000-000000000000/locations/eastus/lro_post/location
+    Expected status code: 202
+    Expected no response body
+    
+    Location second poll.
+    Expected verb: GET
+    Expected URL: {endpoint}/subscriptions/00000000-0000-0000-0000-000000000000/locations/eastus/lro_post/location
+    Expected status code: 200
+    Expected response body:
+    ```json
+    {
+      "url": "https://example.org/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/test-rg/providers/Azure.ResourceManager.OperationTemplates/lroResources/lro/files/myfile"
+    }
+    ```
+    """)
+  @useFinalStateVia("location")
+  upload is ArmResourceActionAsync<
+    LroResource,
+    UploadRequest,
+    UploadResult,
+    LroHeaders = ArmCombinedLroHeaders & Azure.Core.Foundations.RetryAfterHeader
+  >;
+
+  #suppress "deprecated" "legacy"
+  #suppress "@azure-tools/typespec-azure-resource-manager/lro-location-header" "legacy"
+  @scenario
+  @scenarioDoc("""
```

Comment:
curious, what's typespec-azure-resourcemanager's non-legacy DELETE? Is it AAO or Location?

Reply:
Emmm, it's `Location`. Let me remove the unnecessary header.

Reply:
removed

Reply:
Can the `#suppress` be removed too?